File: app/src/main/java/dev/olog/basil/presentation/base/NoBackgroundBottomSheet.java
package dev.olog.basil.presentation.base;

import android.content.Context;
import android.view.View;

import com.google.android.material.bottomsheet.BottomSheetDialog;

import androidx.annotation.NonNull;
import dev.olog.basil.R;

public class NoBackgroundBottomSheet extends BottomSheetDialog {

    public NoBackgroundBottomSheet(@NonNull Context context) {
        super(context);
    }

    public NoBackgroundBottomSheet(@NonNull Context context, int theme) {
        super(context, theme);
    }

    public NoBackgroundBottomSheet(@NonNull Context context, boolean cancelable, OnCancelListener cancelListener) {
        super(context, cancelable, cancelListener);
    }

    @Override
    public void setContentView(View view) {
        super.setContentView(view);
        getWindow().findViewById(R.id.design_bottom_sheet).setBackground(null);
    }
}

File: app/src/main/java/dev/olog/basil/presentation/widget/NoBackgroundBottomSheet.kt
package dev.olog.basil.presentation.widget

import android.content.Context
import android.content.DialogInterface
import android.view.View
import android.widget.FrameLayout

import com.google.android.material.bottomsheet.BottomSheetBehavior
import com.google.android.material.bottomsheet.BottomSheetDialog
import dev.olog.basil.R

class NoBackgroundBottomSheet : BottomSheetDialog {

    constructor(context: Context) : super(context) {}

    constructor(context: Context, theme: Int) : super(context, theme) {}

    constructor(context: Context, cancelable: Boolean, cancelListener: DialogInterface.OnCancelListener) : super(context, cancelable, cancelListener) {}

    override fun setContentView(view: View) {
        super.setContentView(view)
        window!!.findViewById<View>(R.id.design_bottom_sheet).background = null
        val bottomSheet = window.decorView.findViewById<View>(R.id.design_bottom_sheet) as FrameLayout
        val mBehavior = BottomSheetBehavior.from(bottomSheet)
        mBehavior.state = BottomSheetBehavior.STATE_EXPANDED
    }
}
